### ISSUE TITLE
ArangoDB [3.2beta3][RocksDB] slow....to...start if compaction was not triggered before shutdown

### ISSUE THREAD

**Issue by frankmayer** [arangodb/arangodb#2717]
## my environment running ArangoDB
I'm using the latest ArangoDB of the respective release series:
- [x] 3.2 Beta3

On this operating system:
  - [x] docker - official docker library

Here is the scenario:
- I batch-imported around 18.000.000 documents into a collection.
- During the import I was noticing the fluctuations in data size on Disk, which were the result of compactions in RocksDB.
- However, after the import was finished - and having let ArangoDB stay idle for a few minutes, I noticed that the size would not get down to what it was supposed to (~ 3.1GB), but instead staid at ~8.5GB. I guess this is due to the default settings not triggering compaction under those circumstances.
- I restarted ArangoDB and noticed heavy disk access for minutes, before ArangoDB finally started up completely. So, the compaction was triggered on startup, but before ArangoDB was finished starting up. This resulted in ArangoDB not accepting connections until compaction finished.

So the question is: Shouldn't ArangoDB start first and let RocksDB compact after complete startup?

Note that I the data files are located in a Hardware Controlled(with  64MB Cache) RAID1 (Mirror) with 2 spinning disks. Not sure if Data would be on SSDs, behavior would be different in this case. But I believe, that even with spinning Disks, that should not happen.

**Comment by frankmayer** [comment]
With Beta3 I have also noticed, that storage space is not being released after collections and/or databases have been removed. At least not immediately or even after a ArangoDB restart. I have not checked this with Beta4 though. It might be that a compaction is not been triggered, though.
Just mentioning this here, as it might have the same reason.

**Comment by jsteemann** [comment]
You are right. File deletion was not triggered properly in beta2 and beta3. This should have been fixed in beta4 (released yesterday).

**Comment by frankmayer** [comment]
Will check with beta4. Thanks
Does this also apply to the original context of this issue?

**Comment by jsteemann** [comment]
Yes, it should.

**Comment by frankmayer** [comment]
I'll check with the same data and get back to you

**Comment by jsteemann** [comment]
Thanks!

**Comment by frankmayer** [comment]
My findings:
- compactions seem to be triggered correctly now, as a little while after the import was finished, I had about the size on disk that I had with my previous tests after ArangoDB's restart (where it hung - waiting for the overdue compaction)
- Database/Collection deletion still seems not to remove all related files. I found several *.sst files in the rocksdb directory that had the timestamp of the creation, filling and removal of a specific database + collection. Are those files marked for later removal or is some cleanup trigger missing here? FYI, when I deleted the test database, I first deleted the collection, checked if space would be released, then deleted the database and checked again. After that I even restarted ArangoDB without the space being returned. I hope this helps
